app/views/works/show.html.erb

```diff
@@ -24,19 +24,16 @@
 <!--/descriptions-->
 
 <!--subnav-->
-<% if logged_in_as_admin? %>
-  <%= render :partial => 'admin/admin_options', :locals => {:item => @work} %>
+<% if AdminModerationPolicy.can_edit_works?(current_admin) %>
```

Comment:
**1. Instance vs class**

I think the semantically correct way of using Pundit is, test against a specific instance if you have it, otherwise test against the class. We test against the class anyway in some cases, as a shortcut, because we know admins have the same permissions for all instances of that class (we should be consistent and catch such cases in code review, though).

**2. `policy` vs static methods taking `current_admin`**

The reason we're using `AdminModerationPolicy#can_edit_works?` is that we cannot use `policy(@work)`. Frustratingly, `policy` in views relies on name matching to figure out which policy class to pick, and unlike `authorize`, you don't have the option to specify a policy class. We also cannot [declare the policy class](https://github.com/varvet/pundit/#manually-specifying-policy-classes) in AdminModerationPolicy, because it was set up to cover both works and comments.

I stopped here when I last reviewed, but you have a good point about consistency. Maybe we should break AdminModerationPolicy into separate policies for works and comments.

Reply:
Ahh, thank you, I get it now (I think).